### PR TITLE
Enable Text to be used by Viewer outside Files app

### PR DESCRIPTION
**Pull request by danxuliu**
Since [Nextcloud 18.0.4 the Files app triggers the `LoadViewer` event](https://github.com/nextcloud/server/pull/19897), so that event can be used to register Text with Viewer instead of doing it through the `OCA\Files::loadAdditionalScripts` event ~~(although the previous way needs to be kept for now just in case Text is run on a previous Nextcloud 18 version)~~.

Besides being more correct for the Files app it also makes possible to use Text when [opening files through Viewer in other apps, like Talk](https://github.com/nextcloud/spreed/pull/2778).

Note, however, that in public share pages Text is still opened using the fallback actions instead of through Viewer, as Viewer currently does not provide the share token of the public shared folder that contains the file to be opened. This will be addressed in a follow up pull request.

~~Also note that currently the _stable18_ branch of server [still reports 18.0.2 as its version](https://github.com/nextcloud/server/blob/048888be873f6ae1a00fd1e441a8dd14d03ad96b/version.php#L35), so if tested on that branch Text is registered twice with Viewer. That will stop happening once the version is bumped in the _stable18_ branch of server to 18.0.4 and Text is registered only due to the `LoadViewer` event.~~

### CODE REVIEW

File: src/files.js [hunk 1]
```diff
@@ -37,14 +37,33 @@ const workspaceEnabled = loadState('text', 'workspace_enabled')
 registerFileCreate()
 
 document.addEventListener('DOMContentLoaded', () => {
+	function filesAppTriggersLoadViewerEvent() {
+		const [major, minor, maintenance] = OC.config.version.split('.').map(x => parseInt(x))
```
Comment (nickvergessen) [inline src/files.js]: just scrap this and make it 19 only, there is a stable18 branch, so master is okay to be 19 only?

**Comment by nickvergessen** [comment]
Fixed the version in https://github.com/nextcloud/server/pull/20195

File: src/files.js (hunk 1 above)
Comment (danxuliu) [reply src/files.js:41]: No idea, [master branch is currently compatible with 18 and 19](https://github.com/nextcloud/text/blob/d2a8ab5f3f9113ae69603efac5c24290d30a15f6/appinfo/info.xml#L30), so time to summon @juliushaertl

Comment (nickvergessen) [reply src/files.js:41]: Text is shipped, so master is 19 and 18 is stable18

Comment (juliushaertl) [reply src/files.js:41]: Yes, let's just do this on master which is 19 only.

Comment (danxuliu) [reply src/files.js]: > master which is 19 only.

[Not really](https://github.com/nextcloud/text/blob/d2a8ab5f3f9113ae69603efac5c24290d30a15f6/appinfo/info.xml#L30) :-P Anyway I have removed the special handling for 18.

Comment (juliushaertl) [reply src/files.js]: Well yes, that should be fixed, but since we always release the text app together with the server I don't see issues with that :P

**Pull request merged**